src/Sulu/Bundle/AdminBundle/Resources/js/components/ColorPicker/ColorPicker.js

```diff
@@ -0,0 +1,163 @@
+// @flow
+import React, {Fragment} from 'react';
+import type {ElementRef} from 'react';
+import {SketchPicker} from 'react-color';
+import {observer} from 'mobx-react';
+import {action, computed, observable} from 'mobx';
+import Input from '../Input';
+import Popover from '../Popover';
+import colorPickerStyles from './colorPicker.scss';
+import './colorPickerGlobal.scss';
+
+type Props = {|
+    name?: string,
+    onBlur?: () => void,
+    onChange: (value: ?string) => void,
+    placeholder?: string,
+    valid: boolean,
+    value: ?string,
+|};
+
+@observer
+export default class ColorPicker extends React.Component<Props> {
+    static defaultProps = {
+        valid: true,
+    };
+
+    @observable value: ?string;
+    @observable showError: boolean = false;
+    @observable popoverOpen: boolean = false;
+    @observable popoverAnchorElement: ?ElementRef<*>;
```

Comment:
This code is repeated quite often, right? Maybe we can refactor that into a separate `PopoverInput` component or some better name. Doesn't necessarily have to be in this PR, but if we see this pattern more often it might be worth a thought.

Reply:
It's not really always exact the same..